src/region.js

```diff
@@ -71,6 +71,14 @@ _.extend(Region.prototype, CommonMixin, {
 
     if (view === this.currentView) { return this; }
 
+    if (view._isAttached && view._attachedTo) {
+      throw new MarionetteError({
+        name: classErrorName,
+        message: 'A "view" can be attached to the DOM only if it prerendered on backend',
```

Comment:
This message is incorrect. Pre-rendered DOM can be handed to the view by the frontend as well.  Also whether or not this view `_isAttached` is irrelevant here.  If you are showing the same view in two regions and the view with the regions isn't attached you're going to have the same leaky problems.  the only check needs to be `view._attachedTo`

Reply:
what about this?
    if (view._isShown) {
      throw new MarionetteError({
        name: classErrorName,
        message: 'Given view is already shown in some marionette Region or CollectionView',
        url: 'marionette.collectionview.html#region-shownView'
      });
    }

Reply:
> This message is incorrect. Pre-rendered DOM can be handed to the view by the frontend as well. Also whether or not this view `_isAttached` is irrelevant here. If you are showing the same view in two regions and the view with the regions isn't attached you're going to have the same leaky problems. the only check needs to be `view._attachedTo`

https://github.com/marionettejs/backbone.marionette/blob/37bb058441b6d045d93e7051f453cce8b4895a5c/test/unit/view.child-views.spec.js#L433

its not clear for me what exactly happens in this beforeEach hook, but it definitely reshowing already showed view and if i will check only for `_isShown` then i get 3 failed tests included this one

could you shed some light what here happens and how i have to change that

Reply:
i moved _isShown clear to _empty it solves this issue,

Reply:
btw, there is another place
```
          beforeEach(function() {
            ...
            anotherView = new AnotherView();
            ...
            myCollectionView.addChildView(anotherView);
          });
          it('should only append the added child', function() {
            this.sinon.stub(myCollectionView, 'attachHtml');

            // Only true if not maintaining collection sort
            myCollectionView.sortWithCollection = false;
            myCollectionView.addChildView(anotherView);
            ...
          });
```
in terms of this PR's change this can not be done any more
and i can't resolve what exactly this test is testing :-/